src/python/gem5/components/boards/riscv_board.py

```diff
@@ -210,14 +216,14 @@ def _setup_pma(self) -> None:
 
     @overrides(AbstractSystemBoard)
     def has_dma_ports(self) -> bool:
-        return False
+        return self.is_fullsystem()
 
     @overrides(AbstractSystemBoard)
     def get_dma_ports(self) -> List[Port]:
-        raise Exception(
-            "Cannot execute `get_dma_ports()`: Board does not have DMA ports "
-            "to return. Use `has_dma_ports()` to check this."
-        )
+        if self.has_dma_ports():
+            return self._dma_ports
+        else:
+            return None
```

Comment:
We should preserve the original functionality. By returning `None` in the case we don't have DMA ports we're breaking that (it's also not specified by the return type that this function may return a `None` value. Id just suggest:


```suggestion
        if not self.has_dma_ports():
           raise Exception(
            "Cannot execute `get_dma_ports()`: Board does not have DMA ports "
            "to return. Use `has_dma_ports()` to check this."
        )
        return self._dma_ports
```

Reply:
[Done](https://github.com/gem5/gem5/pull/2087/files#diff-7cce437d81e063378e32f58b48e55b95a06f835420d36b39b46eeb72b3567a6cR242-R249)